src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -687,6 +687,25 @@ public override void VisitExpressionStatement(ExpressionStatementSyntax node)
 
         public override void VisitLocalDeclarationStatement(LocalDeclarationStatementSyntax node)
         {
+            var invalidDimensions = ArrayBuilder<ExpressionSyntax>.GetInstance();
+            node.Declaration.Type.VisitTypeSyntaxes(type =>
+            {
+                if (type is ArrayTypeSyntax arrayType)
+                {
+                    foreach (var rankSpecifier in arrayType.RankSpecifiers)
+                    foreach (var size in rankSpecifier.Sizes)
+                    {
+                        if (size.Kind() != SyntaxKind.OmittedArraySizeExpression)
+                        {
+                            invalidDimensions.AddRange(rankSpecifier.Sizes);
```

Comment:
>invalidDimensions.AddRange(rankSpecifier.Sizes); [](start = 28, length = 48)

Is there an advantage in collecting sizes into a builder instead of visiting them on the spot?